cli/flags/global/flags.go

```diff
@@ -66,6 +70,41 @@ const (
 	DeprecatedTfLogJSONFlagName            = "tf-logs-to-json"
 )
 
+// NewFlagsWithDeprecatedMovedFlags returns global flags along with flags that have been moved to other commands and hidden from CLI help.
+func NewFlagsWithDeprecatedMovedFlags(opts *options.TerragruntOptions) cli.Flags {
+	globalFlags := NewFlags(opts, nil)
+
+	// Since the flags we take from the command may already have deprecated flags,
+	// and each time we call `commands.New` we define them again (deprecated flags),
+	// we need to clear `Strict Controls` to avoid them being displayed and causing duplicate warnings, for example, log output:
+	//
+	// WARN The `--terragrunt-no-auto-init` global flag is moved to `run` command and will be removed from the global flags in a future version. Use `--run --no-auto-init=false` instead.
+	// WARN The `--terragrunt-no-auto-init` flag is deprecated and will be removed in a future version. Use `--no-auto-init=false` instead.
```

Comment:
```suggestion
	// WARN The `--terragrunt-no-auto-init` flag is deprecated and will be removed in a future version of Terragrunt. Use `--no-auto-init` instead.
```

Note the second part of the change: users should set the value to `--no-auto-init`, not `--no-auto-init=false`. Even if it works, it's a double-negative and that's confusing.

Reply:
Regarding `=false`, yes I fixed it in this PR. I just took this log when running on the main branch. 

Regarding ` of Terragrunt`, We need to change it [here](https://github.com/gruntwork-io/terragrunt/blob/6bf5e1ce7617a758383d6fb99fc63d3f56864bf1/internal/strict/controls/deprecated_flag_name.go#L43-L44) first.